core/blockchain.go

```diff
@@ -250,20 +254,51 @@ func (bc *BlockChain) VerifyNonce(accountIndex int64, nonce int64) error {
 			return err
 		}
 		if nonce != expectNonce {
-			return errors.New("invalid Nonce")
+			return errors.New("invalid nonce")
 		}
 	} else {
 		pendingNonce, err := bc.Statedb.GetPendingNonce(accountIndex)
 		if err != nil {
 			return err
 		}
 		if pendingNonce != nonce {
-			return errors.New("invalid Nonce")
+			return errors.New("invalid nonce")
 		}
 	}
 	return nil
 }
 
+func (bc *BlockChain) VerifyGas(accountIndex, gasAccountIndex, gasFeeAssetId int64, gasFeeAssetAmount *big.Int) error {
+	gasAccountConfig, err := bc.ChainDB.SysConfigModel.GetSysConfigByName(types2.GasAccountIndex)
```

Comment:
I think gasAccountIndex and gasAsset can get from db once and then saved in memory, so there is no need to get gasAccountConfig and gasAssetConfig for each tx?

Reply:
Agree.